### ISSUE TITLE
Failed to get SSH session: EOF flake on CoreOS clusters

### ISSUE THREAD

**Issue by xmudrii** [kubermatic/kubeone#669]
Trying to run `kubeone install` or `kubeone reset` on a CoreOS host often fails with the following error:

```
time="01:11:17 UTC" level=info msg="Copying Kubeconfig to home directory…" node=15.188.57.146
time="01:11:17 UTC" level=info msg="Copying Kubeconfig to home directory…" node=35.180.191.70
time="01:11:17 UTC" level=info msg="Copying Kubeconfig to home directory…" node=35.180.121.84
time="01:11:17 UTC" level=error msg=": failed to get SSH session: EOF" node=35.180.121.84
time="01:11:18 UTC" level=warning msg="Task failed, retrying…"
```

This error is a flake, i.e. it doesn't happen each time and it's unpredictable.

CoreOS is using a socket for SSH instead of an agent (like on Ubuntu and CentOS). Due to some problem with `systemd`, it seems that the socket is unreliable and often fails. https://github.com/coreos/bugs/issues/2181 has some details about the issue.

As a workaround, we can switch to the agent and it seems to fix the problem. [CoreOS docs](https://coreos.com/os/docs/latest/customizing-sshd.html) has instructions how to do it.

One solution is to use the user data to provide the ignition config to disable the socket and enable `sshd`, and another one is to run the appropriate `systemd` commands over SSH.

For the second approach, we are currently blocked by https://github.com/kubermatic/kubeone/issues/652.

**Action Items:**

* [ ] Update documentation to describe how to solve this problem
* [ ] Make E2E tests use the hook approach to fix this problem

**Comment by xmudrii** [comment]
CoreOS is deprecated and we still didn't check is this problem present for alternatives. If the problem appears again, we'll create a new issue.
/close
